sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryOptionsExtensions.kt

```diff
@@ -50,7 +48,7 @@ internal fun CocoaSentryOptions.applyCocoaBaseOptions(options: SentryOptions) {
         sdk?.set("packages", packages)
 
         event?.sdk = sdk
-        event
+        dropKotlinCrashEvent(event as NSExceptionSentryEvent?) as SentryEvent?
```

Comment:
since `beforeSend` is based on the lambda result we have to put `dropKotlinCrashEvent` at the end.